packages/std/extra/project_version/format.bri

```diff
@@ -0,0 +1,139 @@
+import { PROJECT_VERSION_SEMVER_MATCH } from "./index.bri";
+
+/**
+ * All the possible project version format type.
+ */
+enum ProjectVersionFormatType {
+  DOT = ".",
+  DASH = "-",
+  UNDERSCORE = "_",
+}
```

Comment:
#nonblocking

At some point, I think I want to enable the [`--erasableSyntaxOnly`](https://devblogs.microsoft.com/typescript/announcing-typescript-5-8/#the---erasablesyntaxonly-option) TypeScript option, so I'd prefer to avoid enums when possible

(I also just generally think enums have a few footguns compared to other ways to write the same thing, so I've avoided them for a long time too)

```suggestion
/**
 * All the possible project version format type.
 */
const ProjectVersionFormatType = {
  DOT: ".",
  DASH: "-",
  UNDERSCORE: "_",
} as const;
```

This is a minor nitpick though and it would be easy to tweak in the future so I'm fine with leaving it as a future change too

Reply:
I'll do it, no problem. I didn't know about the [--erasableSyntaxOnly](https://devblogs.microsoft.com/typescript/announcing-typescript-5-8/#the---erasablesyntaxonly-option) TypeScript option. Indeed, let's make the activation of this option less painful by preparing the code for it 😅